### DISCUSSION TITLE
Why is an object's ID random by default rather than a string converted from the name?

### DISCUSSION THREAD

**Discussion by ZeroAurora** [appwrite/appwrite#8229] [Q&A]
I started using Appwrite to build an app recently and noticed that nearly every object id is randomly chosen by default.
By testing on Appwrite Cloud, I understand that orgnization names and project names have chance to collide. But it's not the case for objects like databases, collections and more. These objects are isolated per-project and won't collide with objects in other projects.
I would recommend the sane default id for these objects to be a converted id from the name, i.e. convert Unicode characters to equivalents and whitespaces to hyphens etc.. But please tell me if this is a choice by design. Thank you!

**Comment by wellkilo** [comment]
Random IDs are a deliberate safe default, but they are not mandatory. For resources whose API accepts an ID, you can supply a stable custom ID yourself instead of `ID.unique()`. Appwrite's validation permits `a-z`, `A-Z`, `0-9`, `.`, `-`, and `_`, with a maximum length of 36 and no leading special character.

For example:

```ts
const slug = 'customer-orders'

await databases.createCollection(
  databaseId,
  slug,          // custom collection ID
  'Customer orders',
)
```

I would not make an automatic name-to-ID conversion the platform default, for a few reasons:

- names are mutable display labels, while IDs are commonly used in code/config and should remain stable across renames;
- transliteration/Unicode normalization is locale-dependent and can create collisions;
- two display names can normalize to the same slug;
- predictable IDs may leak business naming and make enumeration easier (permissions still protect data, but random defaults avoid creating that expectation);
- generated resources and concurrent creation need a collision-safe default.

The usual pattern is therefore: use `ID.unique()` unless a stable semantic ID is useful, and if it is, run an application-owned slug/validation function and handle the conflict explicitly. Do not derive IDs from mutable names unless renames are intentionally decoupled from IDs.

Current Appwrite endpoint descriptions explicitly state “choose a custom ID or generate a random ID with `ID.unique()`,” and duplicate-ID errors recommend the same split. If this explains the design and the supported alternative, please mark it as accepted.
